### ISSUE TITLE
OSS detection

### ISSUE THREAD

**Issue by forslund** [MycroftAI/mimic#47]
After setting up a virtual machine without alsa or portaudio OSS is detected as audio despite that no OSS support actually exist. And when running mimic the error message `"oss_audio: failed to open audio device /dev/dsp` is shown.

On modern linux boxes OSS can still be used, there is an alsa compatibility layer and Arch linux has a guide for installing OSS as a native driver. If we intend to keep the OSS support (for linux) some more advanced detection than

```
AC_CHECK_HEADER(sys/soundcard.h,
              [AUDIODRIVER="oss"
               AUDIODEFS=-DCST_AUDIO_LINUX])
```
is needed.

One possibility is to check if `/dev/dsp` actually exists. It is created by the alsa-oss compatibility layer and is the hard coded sound deviced used by mimic's OSS layer. It does feel a bit dirty...

**Comment by zeehio** [comment]
As far as I know, OSS audio was deprecated in the Linux kernel in favor of ALSA. However I believe OSS audio is still used in FreeBSD.

It is reasonable to check for /dev/dsp in the configure script. Maybe even disable OSS if Linux is detected... and print a big warning if there is no detected audio module.

**Comment by forslund** [comment]
There are separate tests for Linux and FreeBSD so it would be rather simple to remove it. I'll post a PR with this.

I might need to get a BSD vm running so we can ensure that OSS is actually working as it should on that platform.

**Comment by zeehio** [comment]
Fixed in #48
